src/ISA/ISA/ARCTypes/CompositeCell.fs

```diff
@@ -1,6 +1,67 @@
 ﻿namespace ISA
 
+// TODO: The design principles say we should use Erase for js compilation. This would also 
+
+//[<Erase>]
+//[<AttachMembersAttribute>]
+// FABLE: Erased unions with multiple cases cannot have more than one field: Test.CompositeCell
+//type CompositeCell = 
+//    | Term of string
+//    | Freetext of {| t: string; f: float |}
+//    | WithUnit of {| t: string; b: bool |}
+
+//    member this.getString =
+//        match this with
+//        | Term s -> s
+//        | Freetext t -> t.t
+//        | WithUnit t -> t.t
+
+//let test_t = Term ("Test me")
+//let test_f = Freetext {|t = "Test me"; f = 12.|}
+
+//test_t.getString |> printfn "%A" // Does return "undefined"
+
+// if we erase we cannot use members
+
 type CompositeCell = 
     | Term of OntologyAnnotation
-    | Freetext of string
-    | WithUnit of string*OntologyAnnotation
\ No newline at end of file
+    | FreeText of string
+    | Unit of string*OntologyAnnotation
```

Comment:
Not sure about this. I see that "WithUnit" sounds weird but "Unit" is not correct either way. People might assume this CompositeCell is just the unit.

Reply:
What about `Measured`?

Reply:
Remember that this is only seen by developers. A vague but easily accessible/memorable term might be better.